provider/provider.go

```diff
@@ -325,6 +334,18 @@ func (p *dockerNativeProvider) Create(ctx context.Context, req *rpc.CreateReques
 		return nil, errors.Wrapf(err, "malformed resource inputs")
 	}
 
+	// if we're in preview mode and buildOnPreview is set to false, we return the inputs
+	if req.GetPreview() && !inputs["buildOnPreview"].BoolValue() {
+		return &rpc.CreateResponse{
+			Properties: req.GetProperties(),
+		}, nil
+	}
+	// buildOnPreview needs all inputs to be resolved. Return error if trying to build on preview and there are Unknowns.
+	if req.GetPreview() && inputs["buildOnPreview"].BoolValue() && inputs.ContainsUnknowns() {
+		return nil, errors.New("cannot build on preview with unresolved inputs. " +
```

Comment:
It feels like this should be a warning instead of an error. If we have agreed that this should error in the past, please ignore this comment.